### ISSUE TITLE
Is pride and diligence really required?

### ISSUE THREAD

**Issue by zyphlar** [Widdershin/programmers-oath#59]
I don't see it in the Hippocratic and I have the same problem with its mention in the Engineer's oath. Pride is the first Deadly Sin, it can inspire some positive things but also some negative things. Seems to me that its function is to foster a guild mentality more than anything. Pride is at odds with humility.

Seems to me that the existing oath covers:

0. Ethics/integrity
1. Respectfulness
2. Empathy
3. Humility
4. Privacy/trustworthy
5. More humility for severe situations
6. Humanity
7. Unforseen consequences
8. Diligence/vigilance/devotion/dependability/discipline/fortitude/loyalty/perseverance/responsibility
9. More humility

So as implied, I would suggest replacing number 8 with some of the words I used (taken from a generic list of virtues, where pride is notably absent.)

**Comment by mo-g** [comment]
As an atheist, I don't see the relevance of 'Deadly Sins' to the argument.

More practically, "Taking pride in ones' work" has a very different meaning to "being Prideful". It means a dedication to doing ones best. Is this a case of interpretation?

**Comment by zyphlar** [comment]
Yes. If you ignore the "deadly sins" aspect I think my comment still stands: colloquial interpretations of the word "pride" are at odds with humility. If you mean a dedication to quality, or any of the other synonyms I listed, say that. Many societies have weird toxic ideas around pride and diligence.

**Comment by Thrilleratplay** [comment]
Larry Wall says there are three great virtues of a programmer: Laziness, Impatience and Hubris.

* Laziness: The quality that makes you go to great effort to reduce overall energy expenditure. It makes you write labor-saving programs that other people will find useful and document what you wrote so you don't have to answer so many questions about it.
* Impatience: The anger you feel when the computer is being lazy. This makes you write programs that don't just react to your needs, but actually anticipate them. Or at least pretend to.
* Hubris: The quality that makes you write (and maintain) programs that other people won't want to say bad things about.

Reducing the oaths down to one or two words may help one remember them, but just like saying laziness, impatience and hubris are virtues, mean can be lost or confused.

**Comment by Widdershin** [comment]
Wow, I hadn't realized that pride and diligence were such loaded terms across different cultures.

To be clear, the sentiment underlying this tenet is effectively: "I will do a good job."

Maybe that should be the wording of the tenet. What do y'all think? Too plain/lacking nuance or just right?

**Comment by zyphlar** [comment]
I'd go with something like "Do what's necessary to ensure high-quality work." I notice quality is mentioned in other issues, maybe this can be wrapped into that.

At the end of the day these are all elaborating on "be good, not bad," so figuring out what it is we're really trying to communicate and whether these are aspirational, or enforceable standards, would be good. Quality is subjective. But where I'm from at least ensuring quality won't be misinterpreted as forming industry cliques or working yourself to death...

**Comment by vassudanagunta** [comment]
How about: 

*I will be diligent and take ownership of my work's flaws as much as I do its feats.*

**Comment by Widdershin** [comment]
> Do what's necessary to ensure high-quality work.

Right now for me that would potentially mean doing hundred hour weeks with no additional compensation.

I can't feasibly sign any oath that requires me to do my utmost to ensure quality, because sometimes the reality of business means that quality has to be sacrificed in the short term.

At the end of the day, I can't see why being diligent and taking pride in our work is a bad thing. I could understand pride in isolation being considered bad, but to take pride in one's work to me is to put yourself into what you do, and to hold yourself to a standard.

**Comment by pachunka** [comment]
> At the end of the day, I can't see why being diligent and taking pride in our work is a bad thing. I could understand pride in isolation being considered bad, but to take pride in one's work to me is to put yourself into what you do, and to hold yourself to a standard.

I know the thread might be all-done, but fwiw: Completely agreed.
